lib/stat_tracker.rb

```diff
@@ -120,17 +122,50 @@ def games_won_per_team_for(season_id)
     end
   end
 
-  def shots_to_goals_ratio_per_team
-    total_shots_per_team.merge(total_goals_per_team){|team_id, shots, goals| (shots.to_f / goals).round(2)}
-  end
-
   def total_games_per_team_for(season_id)
     @game_teams.reduce(Hash.new(0)) do |result, game_team|
       result[game_team.team_id] += 1 if game_team.season == season_id
       result
     end
   end
 
+  def all_coaches
+   @game_teams.map {|game_team| game_team.head_coach}.uniq
+  end
+
+  def create_hash_with_team_games_by_coach(season_id)
+     all_coaches.reduce({}) do |coaches_with_games, coach|
+       coaches_with_games[coach] = game_teams_that_season_by_coach(coach, season_id)
+       coaches_with_games
+    end
+  end
+
+  def game_teams_that_season_by_coach(coach, season_id)
+    @game_teams.find_all do |game_team|
+      game_team.game_id.to_s[0..3] == season_id.to_s[0..3] && game_team.head_coach == coach
+    end
+  end
+
+  def number_of_games_by_coach(season_id)
+    create_hash_with_team_games_by_coach(season_id).transform_values do |game_teams|
+      game_teams.length
+    end
+  end
+
+  def find_all_wins_by_coach(season_id)
+    create_hash_with_team_games_by_coach(season_id).transform_values do |game_teams|
+      (game_teams.find_all {|game| game.result == "WIN"}).length
+    end
+  end
+
+  def percent_wins_by_coach(season_id)
+    percent_wins = {}
+    find_all_wins_by_coach(season_id).map do |coach, num_wins|
+      percent_wins[coach] = (num_wins.to_f / number_of_games_by_coach(season_id)[coach]).round(2)
+    end
+     percent_wins
+  end
+
```

Comment:
@sarynm12 I tried adding a conditional statement to this one that you're running into errors with. I added if percent_wins[coach] = 0, then 0.0 and else, do the calculation you have on line 164. Then in the tests I changed the NaN to 0.0 and it passed. Would that help solve your issues here?

Reply:
Yeah, that will do it! I'll make changes push back up. Thanks Leah!

Reply:
I found a way to just get rid of the coaches that had no wins! About to push back up! See what you think :)

Reply:
Nice!